update_promotable_jobs.py

```diff
@@ -9,16 +11,26 @@
 
 
 def run():
-    job_data = parse_job_data()
-    jobs = determine_jobs(job_data)
-    for job in jobs:
-        cluster = job_data[job].get('cluster', '')
-        for environment in ['staging', 'production']:
-            update_job_promotion(job, environment, cluster)
-
-def parse_job_data():
+    job_data = parse_job_data(['jobs/app-jobs.yml','jobs/java-docker-jobs.yml'])
+    try:
+       jobs = determine_jobs(job_data)
+       xml_config_file = "jobs/promotion-config.xml"
+       for job in jobs:
+           cluster = job_data[job].get('cluster', '')
+           for environment in ['staging', 'production']:
+               update_job_promotion(job, environment, cluster, xml_config_file)
+    finally:
```

Comment:
Without it, the first block (that looks for non-EP jobs) would raise an exception when you want to process an EP job, and the rest of the flow wouldn't be processed.  It was the way that I figured out to get the EP block to run without removing the "raise" from the first block.  Hope that makes sense.  I'm a python beginner.  I can talk after 2pm eastern if you want.